basxconnect/core/views/person/person_browse_views.py

```diff
@@ -31,6 +39,112 @@ def bulkrestore(request, qs):
+                    person.tags.add(tag)
+                else:
+                    person.tags.remove(tag)
+                person.save()
+            return HttpResponseRedirect(reverse_model(models.Person, "browse"))
+
+    form = BulkTagOperationForm()
+    count = len(persons)
+    if operation == "add":
+        header = (
+            ngettext_lazy(
+                "Add tag to %(count)d person",
+                "Add tag to %(count)d persons",
+                count,
+            )
+            % {"count": count}
+        )
+    else:
+        header = (
+            ngettext_lazy(
+                "Remove tag from %(count)d person",
+                "Remove tag from %(count)d persons",
+                count,
+            )
+            % {"count": count}
+        )
+    tags_vocabulary_id = Vocabulary.objects.filter(slug="tag").first().id or ""
+    return layout.render(
+        request,
+        import_string(django.conf.settings.DEFAULT_PAGE_LAYOUT)(
+            menu.main,
+            bread.layout.forms.Form(
+                form,
+                hg.H3(header),
+                hg.DIV(
+                    hg.DIV(bread.layout.forms.FormField("tag")),
+                    hg.If(
+                        operation == "add",
+                        hg.DIV(
+                            modal_with_trigger(
+                                Modal.with_ajax_content(
+                                    heading=_("Create new tag"),
+                                    url=ModelHref(
+                                        Term,
+                                        "add",
+                                        query={
+                                            "vocabulary": tags_vocabulary_id,
+                                            "asajax": True,
+                                        },
+                                    ),
+                                    submitlabel=_("Save"),
+                                ),
+                                Button,
+                                _("Create new tag"),
+                                buttontype="ghost",
+                                style="margin-bottom: 2rem;",
+                                icon="add",
+                            ),
+                        ),
+                    ),
+                    style="display:flex;align-items:end;",
+                ),
+                layout.forms.helpers.Submit(_("Submit")),
+            ),
+        ),
+    )
+
+
+def bulkaddtag(request, qs):
+    return _redirect_to_tag_operation(qs, "add")
+
+
+def bulkremovetag(request, qs):
+    return _redirect_to_tag_operation(qs, "remove")
+
+
+def _redirect_to_tag_operation(qs, operation):
+    return HttpResponseRedirect(
+        reverse_model(
+            models.Person,
+            "bulk-tag-operation",
+            query={
+                "operation": operation,
+                "persons": ",".join([str(person.id) for person in qs]),
```

Comment:
In Django we can normally pass "arrays" via GET query like this:

    ?list=item1&list=item2&list=item3

GET queries are not normal dicts, they allow multiple values with the same keys. Django then parses that into a list. And the serialization should work the same, the other way around. I.e. I think it should be possible to just pass ```[p.id for p in qs]``` and then get a list in ```request.GET["persons"]``` in the target view function. This is also "HTTP-standardized", as a default-multiselect form element will generate the form data that way.
I think comma-separation is sensible too, but (re-)using a standard mechanism is always nice. In theory you could even define a form-class inside the target-function which would automatically convert the GET-query into a a form with a field of type ModelChoiseField.